activerecord/lib/active_record/persistence.rb

```diff
@@ -285,11 +285,12 @@ def instantiate(attributes, column_types = {}, &block)
       # it is preferred to use {update_all}[rdoc-ref:Relation#update_all]
       # for updating all records in a single query.
       def update(id = :all, attributes)
-        if id.is_a?(Array)
+        case id
+        when Array
```

Comment:
It looks like a cosmetic change.
That is the reason that originaly `if id.is_a?(Array)` was used is `elsif id == :all` didn't exist before.